Bottom: 7157432018a089cd4b09c93cc65932922db80ac9
Top:    c677165ae72721d4455590ed99d7a5b13daeb474
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 20:30:52 +0000

Workaround crash where contour is passed as NULL to poly_PreContour













---

diff --git a/src/polygon.c b/src/polygon.c
index 1999b43..471b91c 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -253,6 +253,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
